eng/tools/versioning/set-dev.js

```diff
@@ -23,55 +23,196 @@ const path = require("path");
+
+    console.log(`checking ${depName}:${depVersionRange}...`);
+
+    // If the dependency isn't part of the Rush workspace, skip it
+    if (!rushPackages[depName]) {
+      return rushPackages;
+    }
+
+    // Compare the dependency version range with the package's current version
+    const packageVersion = rushPackages[depName].json.version;
+
+    console.log(`version in package's dep = ${depVersionRange}`);
+    console.log(`dep's version = ${packageVersion}`);
+    const parsedPackageVersion = semver.parse(packageVersion);
+    const parsedDepMinVersion = semver.minVersion(depVersionRange);
+    // If the dependency range is satisfied by the package's current version,
+    // replace it with an exact match to the package's new version
+    if (
+      parsedDepMinVersion.major == parsedPackageVersion.major &&
+      parsedDepMinVersion.minor == parsedPackageVersion.minor &&
+      parsedDepMinVersion.patch == parsedPackageVersion.patch &&
+      rushPackages[depName].newVer !== undefined
+    ) {
+      // Commit the dependency version to the JSON document
+      dependencySection[
+        depName
+      ] = `^${parsedPackageVersion.major}.${parsedPackageVersion.minor}.${parsedPackageVersion.patch}-dev`;
+    }
+  }
+  return rushPackages;
+};
+
+const makeInternalDependenciesConsistent = (rushPackages, package,depName) => {
+  console.log("makeInternalDepConsistent");
+  console.log("package = "+ package);
+  console.log("depName="+depName);
+  console.log("checking dependencies ..");
+
+  rushPackages = makeDependencySectionConsistentForPackage(rushPackages, rushPackages[package].json.dependencies,depName);
+
+  console.log("checking devDependencies ..");
+  rushPackages = makeDependencySectionConsistentForPackage(rushPackages, rushPackages[package].json.devDependencies,depName);
+
+  console.log("checking peerDependencies ..");
+  rushPackages = makeDependencySectionConsistentForPackage(rushPackages, rushPackages[package].json.peerDependencies,depName);
+  return rushPackages;
+};
+
+async function main(argv) {
+  const buildId = argv["build-id"];
+  const repoRoot = argv["repo-root"];
+  const service = argv["service"];
+
+  var rushPackages = await versionUtils.getRushPackageJsons(repoRoot);
+
+  let targetPackages = [];
+  for (const package of Object.keys(rushPackages)) {
+    if (
+      ["client", "core"].includes(rushPackages[package].versionPolicy) &&
+      rushPackages[package].projectFolder.startsWith(`sdk/${service}`)
+    ) {
+      targetPackages.push(package);
+    }
+  }
+
+  // Set all the new versions
+  console.log(`Updating packages with build ID ${buildId}`);
+  for (const package of targetPackages) {
+    console.log("package updated = ");
+    console.log(package);
+    rushPackages = updatePackageVersion(rushPackages, package, buildId);
+    console.log(rushPackages[package].newVer);
+  }
+
+  // Update any references to internal projects with the new versions
```

Comment:
We can do do these in the same for loop.

Reply:
done